.github/workflows/build_and_test.yml

```diff
@@ -80,15 +80,31 @@ jobs:
 
   wasm-build:
     name: run wasm build script
-    runs-on: ubuntu-latest
     strategy:
       matrix:
-        toolchain:
-          - stable
+        os: [ubuntu-latest, macos-latest-large, macos-latest]
+    runs-on: ${{ matrix.os }}
 
     env:
       RUSTFLAGS: '-D warnings -F unsafe-code'
 
     steps:
       - uses: actions/checkout@v4
-      - run: cd ./cedar-wasm && cargo install wasm-pack && TEST_TS=1 ./build-wasm.sh
+      - name: update llvm (macOS)
+        if: ${{ matrix.os  == 'macos-latest-large' }}
+        run: brew install llvm && echo "/usr/local/opt/llvm/bin" >> $GITHUB_PATH
+      - name: update llvm (macOSarm64)
+        if: ${{ matrix.os  == 'macos-latest' }}
+        run: brew install llvm && echo "/opt/homebrew/opt/llvm/bin" >> $GITHUB_PATH
+      - name: install typescript
+        if: ${{ matrix.os  != 'ubuntu-latest' }}
+        run: npm install -g typescript
+      - name: install wasm-pack
+        run: cargo install wasm-pack
+      - name: build wasm (ubuntu)
+        if: ${{ matrix.os  == 'ubuntu-latest' }}
+        run: cd ./cedar-wasm && TEST_TS=1 ./build-wasm.sh
+      - name: build wasm (macOS)
+        if: ${{ matrix.os  != 'ubuntu-latest' }}
+        # testing generated ts code fails on macOS
```

Comment:
any ideas why?

Reply:
Yes. It will generate code like the following,

```typescript
export type ExprNoExt = { Value: CedarValueJson } | { Var: Var } | { Slot: string } | { !: { arg: Expr } } | { neg: { arg: Expr } } | { ==: { left: Expr; right: Expr } } | { !=: { left: Expr; right: Expr } } | { in: { left: Expr; right: Expr } } | { <: { left: Expr; right: Expr } } | { <=: { left: Expr; right: Expr } } | { >: { left: Expr; right: Expr } } | { >=: { left: Expr; right: Expr } } | { &&: { left: Expr; right: Expr } } | { ||: { left: Expr; right: Expr } } | { +: { left: Expr; right: Expr } } | { "-": { left: Expr; right: Expr } } | { *: { left: Expr; right: Expr } } | { contains: { left: Expr; right: Expr } } | { containsAll: { left: Expr; right: Expr } } | { containsAny: { left: Expr; right: Expr } } | { .: { left: Expr; attr: SmolStr } } | { has: { left: Expr; attr: SmolStr } } | { like: { left: Expr; pattern: PatternElem[] } } | { is: { left: Expr; entity_type: SmolStr; in?: Expr } } | { "if-then-else": { if: Expr; then: Expr; else: Expr } } | { Set: Expr[] } | { Record: Record<string, Expr> };
```

Note that there are symbols unquoted like `+`.

Reply:
Could possibly be due to gnu and bsd `sed` using different regex dialects

Reply:
You're so much better at computers than I am. I think it should be root cause.

Reply:
Tried to make it portable.